core/os/os_linux.odin

```diff
@@ -1144,3 +1144,50 @@ ppoll :: proc(fds: []pollfd, timeout: ^unix.timespec, sigmask: ^sigset_t) -> (in
 	}
 	return result, ERROR_NONE
 }
+
+copy_file :: proc(src: string, dst: string, overwrite: bool = false) -> Errno {
+	runtime.DEFAULT_TEMP_ALLOCATOR_TEMP_GUARD()
+	src_handle, src_err := open(src)
+	if src_err != ERROR_NONE {
+		return src_err
+	}
+	defer close(src_handle)
+
+	flags := O_WRONLY | O_CREATE
+	if !overwrite {
+		flags |= O_EXCL
+	}
+
+	dst_handle, dst_err := open(dst, flags, 0666)
+	if dst_err != ERROR_NONE {
+		return dst_err
+	}
+	defer close(dst_handle)
+
+	buf, buf_err := make([]u8, 4096, context.temp_allocator)
+	if buf_err != .None {
+		return ENOMEM
+	}
+
+	for {
+		bytes_read, read_err := read(src_handle, buf)
+		if read_err != ERROR_NONE {
+			return read_err
+		} else if (bytes_read == 0) {
+			break;
+		}
+
+		total_written := 0
+		for total_written < bytes_read {
+			bytes_written, write_err := write(dst_handle, buf[total_written:bytes_read])
```

Comment:
Same deal here.

Reply:
No, it looks correct.

It tries to read 4k at a time and `bytes_read` is how many it actually read.
Then `total_written` is reset to `0` each time it tries to write that 4k block, so Tetra's proposal would cause the `buf` slice to go out of bounds.